### ISSUE TITLE
`reportInvalidTypeVarUse` false positive on classes when default value is `...`

### ISSUE THREAD

**Issue by DetachHead** [DetachHead/basedpyright#595]
```py
from types import EllipsisType
class Foo[T]:
    def __init__(
        self, default_value: T | EllipsisType = ...
    ) -> None:
        ...
```
```
Type variable "T" may go unsolved if caller supplies no argument for parameter "default_value"
  Provide an overload that specifies the return type when the argument is not supplied
```
[playground](https://basedpyright.com/?typeCheckingMode=all&code=GYJw9gtgBALgngBwKYGcoEsILCGUCiANoegiuigCqJIBQAxoQIYpoBiYYA2pQLoBctKMKgATJMCgB9KegB26GDIAUQkepRJCwADRiJTAK6ElANyaFDSflEpQAPgWKlyVGlAC8UAHS%2B1wgEooAFoAPigAOTA5a391H18gA)

a more realistic example is when the default value is omitted, eg. in a stub file:

```py
class Foo[T]:
    def __init__(self, default_value: T | None = ...) -> None: # error
        ...
```

this error should never be reported on classes, because the generic can be explicitly specified anyway:
```py
foo = Foo[int]()
```
